apps/dashboard/src/components/workflow-editor/schema.ts

```diff
@@ -23,10 +23,12 @@ export const workflowSchema = z.object({
   description: z.string().max(MAX_DESCRIPTION_LENGTH).optional(),
 });
 
-export const stepSchema = z.object({
-  name: z.string().min(1).max(MAX_NAME_LENGTH),
-  stepId: z.string(),
-});
+export const buildStepSchema = (controlsSchema?: z.ZodObject<any>) =>
```

Comment:
Please add a TODO to return all the step schemas from the API. Since we are returning step controls, we should return everything.